lib/iopromise/data_loader.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module IOPromise
+  module DataLoader
+    module ClassMethods
+      def attr_promised_data(*args)
+        @promised_data ||= []
+        @promised_data.concat(args)
+
+        args.each do |arg|
+          self.class_eval("def #{arg};@#{arg}.sync;end")
```

Comment:
I was going to suggest to use `define_method` here which would improve readability of the method implementation. But short research, it looks like methods defined via `class_eval` are significantly faster compared to `define_method` (which uses a `Proc`).

Reply:
I was basing this off other examples I'd seen, but good to know there is a reason for one over the other!